test_conformance/api/test_kernel_private_memory_size.cpp

```diff
@@ -0,0 +1,58 @@
+#include "harness/errorHelpers.h"
+#include "harness/typeWrappers.h"
+#include <iostream>
+#include <vector>
+#include "procs.h"
+
+constexpr char const* ZERO_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer , local ulong x[1]){)"
+    R"(    buffer[0] = x[0]; })";
+
+constexpr char const* ONE_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer ){ )"
+    R"(    volatile __private ulong x[1];)"
+    R"(    buffer[0] = x[0]; })";
+
+constexpr char const* TWO_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer ){ )"
+    R"(    volatile __private ulong x[1];)"
+    R"(    volatile __private ulong y[1];)"
+    R"(    buffer[0] = x[0] = y[0]; })";
+
+static std::vector<char const*> kernels = { ZERO_MEMORY_KERNEL,
+                                            ONE_MEMORY_KERNEL,
+                                            TWO_MEMORY_KERNEL };
+static std::vector<cl_ulong> expected_private_memory = { 0, 8, 16 };
+
+int test_kernel_private_memory_size(cl_device_id deviceID, cl_context context,
+                                    cl_command_queue queue, int num_elements)
+{
+    for (int i = 0; i < kernels.size(); i++)
+    {
+        cl_int err = 0;
+        clProgramWrapper program(
+            clCreateProgramWithSource(context, 1, &kernels[i], nullptr, &err));
+        ASSERT_SUCCESS(err, "clCreateProgramWithSource");
+        err = clBuildProgram(program, 1, &deviceID, nullptr, nullptr, nullptr);
+        ASSERT_SUCCESS(err, "clBuildProgram");
+
+        /* Create the Kernel*/
+        clKernelWrapper kernel(clCreateKernel(program, "private_memory", &err));
+        ASSERT_SUCCESS(err, "clCreateKernel");
+
+
+        cl_ulong size = -1; // If clGetKernelWorkGroupInfo passes the value
```

Comment:
Could we use `CL_ULONG_MAX` here please? Using -1 will trigger some errors/warnings in some compilers.

Reply:
Yes, this has been updated in the latest commit. Is this what you were looking for?